src/GitLabApiClient/FileClient.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Web;
+using GitLabApiClient.Internal.Http;
+using GitLabApiClient.Internal.Paths;
+using GitLabApiClient.Internal.Queries;
+using GitLabApiClient.Internal.Utilities;
+using GitLabApiClient.Models.Files.Requests;
+using GitLabApiClient.Models.Files.Responses;
+
+namespace GitLabApiClient
+{
+    public sealed class FilesClient
+    {
+        private readonly GitLabHttpFacade _httpFacade;
+        private readonly FileQueryBuilder _fileQueryBuilder;
+
+        internal FilesClient(GitLabHttpFacade httpFacade, FileQueryBuilder fileQueryBuilder)
+        {
+            _httpFacade = httpFacade;
+            _fileQueryBuilder = fileQueryBuilder;
+        }
+        public async Task<File> GetAsync(ProjectId projectId, string filePath, Action<FileQueryOptions> options = null)
```

Comment:
Just tested the `file` api, as `ref` option is mandatory I believe it is best to have it as a parameter and not as an option, that is somewhat hidden.

The Files API only have a `ref` query option, doc even says it is required
https://docs.gitlab.com/ee/api/repository_files.html#get-file-from-repository

sadly enough this is not true for the `tree` api: https://docs.gitlab.com/ee/api/repositories.html it will assume `default branch` if `ref` is not provided. Gotta love GitLab's inconsistent API 😆

So I suggest for now we remove the `FileQueryOptions` and just switch to `projects/{projectId}/repository/files/{filePath.UrlEncode()}?ref={reference}`

Reply:
I think it would be nice to provide an option to convert `File` to `Content` by handling the decoding of base64.